functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+function Restore-DbaBackup
+{
+<#
+.SYNOPSIS 
+Restores a SQL Server Database from a set of backupfiles
+
+.DESCRIPTION
+Scans a given folder for Full, Differential and Log backups. 
+OR
+Takes a set of folder arrays and processes them for Full, Differential and Log backups.
+
+These are then filtered and restored to a specified SQL Server intance and file location
+
+The backup LSN chain and RecoveryForkID will also be checked to ensure the restore is valid
+
+It can also generate restore scripts, both as part of a restore or as it's only action
+
+.PARAMETER SqlServer
+The SQL Server instance. 
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. 
+
+.PARAMETER Path
+Path to SQL Server backup file. This can be a full, differential or log backup file.
+
+.PARAMETER Files
+A Files object(s) containing SQL Server backup files. 
+
+.PARAMETER RestoreLocation
+Path to restore the SQL Server backups to on the target inance
+
+.PARAMETER RestoreTime
+Specify a DateTime object to which you want the database restored to. Default is to the latest point available 
+
+.PARAMETER OlaStyle
+Switch to indicate the backup files are in a folder structure as created by Ola Hallengreen's maintenance scripts
+
+.PARAMETER FileDirectory
+Switch to indicate the backup files just exist in a folder (this is the standard)
+	
+.PARAMETER FileList
+Returns detailed information about the files within the backup
+
+.PARAMETER DatabaseName
+Name to restore the database under
+
+.PARAMETER NoRecovery
+Indicates if the database should be recovered after last restore. Default is to recover
+
+.PARAMETER ReplaceDatabase
+Switch indicated is the restore is allowed to replace an existing database.
+
+.PARAMETER Scripts
```

Comment:
Yes, I pinched the SQLPS version.

OutScript works for me

Scripts are just dumped to output atm